packages/components/src/ui/create-styles/create-style-system/create-core-element.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * External dependencies
+ */
+import isPropValid from '@emotion/is-prop-valid';
+
+/**
+ * WordPress dependencies
+ */
+import { useMergeRefs } from '@wordpress/compose';
+import { forwardRef, createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useHydrateGlobalStyles } from '../hooks';
+import {
+	INTERPOLATION_CLASS_NAME,
+	REDUCED_MOTION_MODE_ATTR,
+} from './constants';
+import {
+	DEFAULT_STYLE_SYSTEM_OPTIONS,
+	getInterpolatedClassName,
+} from './utils';
+
+const shouldForwardProp = isPropValid;
+
+const defaultOptions = DEFAULT_STYLE_SYSTEM_OPTIONS;
+
+/**
+ * @typedef CreateCoreElementOptions
+ * @property {import('create-emotion').ObjectInterpolation<any>} baseStyles The baseStyles from the Style system.
+ * @property {import('../create-compiler').Compiler} compiler The injectGlobal from the Style system's compiler.
+ * @property {import('./generate-theme').GenerateThemeResults} globalStyles The globalStyles from the Style system.
+ */
+
+/**
+ * Creates the core styled elements for the Style system. These elements are
+ * an in-between of Emotion's <div css="" /> solution and a styled.div``
+ * solution.
+ *
+ * createCoreElement is a super light-weight higher-order wrapper that
+ * enhances base elements (like `div`, `input`, or even Components) with
+ * features provided by the Style system, such as direct CSS compiling via
+ * the `css` prop.
+ *
+ * A styled element also has built-in baseStyles (which can be adjusted using
+ * the createStyleSystem factory).
+ *
+ * @example
+ * ```jsx
+ * const alwaysBlueDiv = createCoreElement('div', { baseStyles: { background: 'blue' }})
+ * ```
+ *
+ * @template {keyof JSX.IntrinsicElements} TTagName
+ * @param {TTagName} tagName The HTMLElement/React.Component to connect with the Style system.
+ * @param {CreateCoreElementOptions} options Options to custom coreElement styles.
+ * @return {import('./polymorphic-component').PolymorphicComponent<TTagName, {}>} The Style system wrapped HTMLElement/React.Component.
+ */
+export const createCoreElement = ( tagName, options ) => {
```

Comment:
I'm actually wondering whether this is really necessary at all

Reply:
We can talk about it here: https://github.com/WordPress/gutenberg/pull/30509#discussion_r609709604